frontend_tests/run-casper

```diff
@@ -39,73 +39,85 @@ parser.add_option('--remote-debug',
     except:
         return False
 
-ret = 1
-
-try:
-    # Wait for the server to start up.
-    sys.stdout.write('Waiting for test server')
-    while not server_is_up():
-        sys.stdout.write('.')
-        sys.stdout.flush()
-        time.sleep(0.1)
-    sys.stdout.write('\n')
+def run_tests(realms_have_subdomains, files):
+    test_files = []
+    for file in files:
+        if not os.path.exists(file):
+            file = os.path.join(os.path.dirname(__file__), '../frontend_tests/casper_tests', file)
+        test_files.append(os.path.abspath(file))
 
+    subprocess.check_call('tools/setup/generate-fixtures')
+    server = subprocess.Popen(('tools/run-dev.py', '--test'),
+                          stdout=log, stderr=log)
     remote_debug = ""
     if options.remote_debug:
         remote_debug = "--remote-debugger-port=7777 --remote-debugger-autorun=yes"
-    cmd = "frontend_tests/casperjs/bin/casperjs %s test " % (remote_debug,)
+    cmd = "frontend_tests/casperjs/bin/casperjs %s test --subdomains=%s " % (remote_debug, realms_have_subdomains,)
 
     if test_files:
         cmd += ' '.join(test_files)
     else:
         cmd += 'frontend_tests/casper_tests'
-    print("Running %s" % (cmd,))
-    ret = subprocess.call(cmd, shell=True)
-finally:
-    assert_server_running()
-    server.terminate()
-
-if ret != 0:
-    print("""
-Oops, the frontend tests failed. Tips for debugging:
- * Check the frontend test server logs at frontend_tests/casper_tests/server.log
- * Check the screenshots of failed tests at /tmp/casper-failure*.png
- * Try remote debugging the test web browser as described in docs/testing.rst
-""", file=sys.stderr)
-
-sys.exit(ret)
+
+    ret = 1
+
+    try:
+        # Wait for the server to start up.
+        sys.stdout.write('Waiting for test server')
+        while not server_is_up(server):
+            sys.stdout.write('.')
+            sys.stdout.flush()
+            time.sleep(0.1)
+        sys.stdout.write('\n')
+
+        print("Running %s" % (cmd,))
+        ret = subprocess.call(cmd, shell=True)
+    finally:
+        assert_server_running(server)
+        server.terminate()
+
+    if ret != 0:
+        print("""
+    Oops, the frontend tests failed. Tips for debugging:
+    * Check the frontend test server logs at frontend_tests/casper_tests/server.log
+    * Check the screenshots of failed tests at /tmp/casper-failure*.png
+    * Try remote debugging the test web browser as described in docs/testing.rst
+    """, file=sys.stderr)
+        sys.exit(ret)
+
+# Run tests with REALMS_HAVE_SUBDOMAINS set to False
+os.environ["EXTERNAL_HOST"] = "localhost:9981"
+run_tests(realms_have_subdomains=False, files=args)
+os.environ["EXTERNAL_HOST"] = "zulipdev.com:9981"
+os.environ["SESSION_COOKIE_DOMAIN"] = "zulipdev.com"
+os.environ["REALMS_HAVE_SUBDOMAINS"] = "True"
+# Run tests with REALMS_HAVE_SUBDOMAINS set to True. Only the first test is run.
+run_tests(realms_have_subdomains=True, files=["00-realm-creation.js"])
```

Comment:
I think this should probably intersect the list of test files we run with `realms_have_subdomains` with the list the user specified on the command line -- the current code basically breaks support for running just a single test.